### ISSUE TITLE
Update DOC: examples vault-pki

### ISSUE THREAD

**Issue by mgriffin13** [hashicorp/consul-template#1083]
I would like to propose an update to the ``examples: vault-pki`` documentation.

The existing example states that the following is valid to use when writing a ca.crt file.

```liquid
{{- /* /tmp/ca.tpl */ -}}
{{ with secret "pki/issue/my-domain-dot-com" "common_name=foo.example.com" }}
{{ .Data.ca_chain }}{{ end }}
```

In reality, the ``ca_chain`` field returns an array of all issuing CAs, thus using the example as-is produces an invalid ``ca.crt`` file that begins with the ``[``, like this:
```
[-----BEGIN CERTIFICATE-----
MIIFTTCCAzWgAwIBAgIUbhGwmRe6mky4rd6fBL36pTaUv1cwDQYJK
```

I would propose changing the example to be ``{{ .Data.issuing_ca }}{{ end }}``, or for a more complete example, you could iterate through the ``ca_chain`` array and write all the CAs:

```liquid
{{- /* /tmp/ca.tpl */ -}}
{{ with secret "pki/issue/my-domain-dot-com" "common_name=foo.example.com" }}
{{ range $cert := .Data.ca_chain }}
{{ $cert }}
{{ end }}
{{ end }}
```

**Comment by pearkes** [comment]
Thanks for the feedback @mgriffin13. Noted, but we may not get to this soon. Would happily review a PR for this change if you're able!

Note for others the example in question is [here](https://github.com/hashicorp/consul-template/blob/master/examples/vault-pki.md).

**Comment by brooksgarrett** [comment]
PR submitted since I also ran into this issue. When the issuing CA is a root CA the `ca_chain` had `<no_value>` and caused issues in my case.

**Comment by pearkes** [comment]
@brooksgarrett Thanks! Were you able to test and verify the behavior of the example? Looks good from here..

**Comment by brooksgarrett** [comment]
The sample comes direct from the workaround I used to solve this exact issue. I also examined the Vault response to ensure it was correct.

**Comment by pearkes** [comment]
Great thanks @brooksgarrett merging now...

**Comment by pearkes** [comment]
Fixed in #1090
